src/components/Form/FormProvider.js

```diff
@@ -243,19 +248,34 @@ function FormProvider({validate, formID, shouldValidateOnBlur, shouldValidateOnC
                 // as this is already happening by the value prop.
                 defaultValue: undefined,
                 onTouched: (event) => {
-                    setTouchedInput(inputID);
+                    if (shouldSetTouchedOnBlurOnly) {
+                        setTimeout(() => {
+                            setTouchedInput(inputID);
+                        }, VALIDATE_DELAY);
+                    }
                     if (_.isFunction(propsToParse.onTouched)) {
                         propsToParse.onTouched(event);
                     }
                 },
                 onPress: (event) => {
-                    setTouchedInput(inputID);
+                    if (shouldSetTouchedOnBlurOnly) {
+                        setTimeout(() => {
+                            setTouchedInput(inputID);
+                        }, VALIDATE_DELAY);
+                    }
                     if (_.isFunction(propsToParse.onPress)) {
                         propsToParse.onPress(event);
                     }
                 },
-                onPressIn: (event) => {
-                    setTouchedInput(inputID);
+                onPressOut: (event) => {
+                    // To prevent validating just pressed inputs, we need to set the touched input right after
+                    // onValidate and to do so, we need to delays setTouchedInput of the same amount of time
+                    // as the onValidate is delayed
+                    if (shouldSetTouchedOnBlurOnly) {
```

Comment:
I would give almost anything to avoid adding `setTimeout()`s to the code. It leads to cascades of more and more of them. What other alternatives are there?

Reply:
I totally agree, that we should avoid using `setTimeout` but this code is the mentioned cascade. It was introduced [here](https://github.com/Expensify/App/pull/16444) and it looks like there is no better solution - at least nobody managed to propose anything better [than this](https://github.com/Expensify/App/issues/13623#issuecomment-1442222158)

Reply:
Yea, this was discussed extensively and we didn't come up with any good alternatives. We could reopen the discussion but I wouldn't block this PR on it since this is existing logic in the Form component